docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+
+**Note:** These samples use the `jq` command to parse json. You can run `jq --version` to check if you have it installed. Otherwise, install `jq` or suitably modify the scripts yourself.
+
+### /bin/detect
+
+The `bin/detect script` is responsible for determining whether or not a buildpack should be used to build a chaincode package and launch it. For chaincode as an external service, the script should detect that metadata.json `type` is set to `external`.  The peer invokes detect with two arguments:
+
+```
+bin/detect CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR
+```
+
+A typical `detect` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$#" -ne 2 ]; then
+    echo "Expected 2 directories got $#"
+    exit 2
+fi
+
+#check if the "type" field is set to "external"
+if [ "$(jq -r .type "$2/metadata.json")" == "external" ]; then
+    exit 0
+fi
+
+exit 1
+```
+Recall that the metadata.json file should contain the following keys:
+
+```json
+{"path":"","type":"external","label":"mycc"}
+```
+
+### /bin/build
+
+The `bin/build` script is responsible for building, compiling, or transforming the contents of a chaincode package into artifacts that can be used by release and run. For chaincode as an external service, the build script copies the `connection.json` to the `BUILD_OUTPUT_DIR`. The peer invokes the build script with three arguments:
+
+```
+bin/build CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR BUILD_OUTPUT_DIR
+```
+
+A typical `build` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$#" -ne 3 ]; then
+    echo "Expected 3 directories got $#"
+    exit 1
+fi
+
+SOURCE=$1
+OUTPUT=$3
+
+#external chaincodes expect connection.json file in the chaincode package
+if [ ! -f "$SOURCE/connection.json" ]  ; then
+    echo "$SOURCE/connection.json not found" > /tmp/err.txt
+    exit 1
+fi
+
+#do more validation here if needed
+
+#simply copy the endpoint information to specified output location
+cp $SOURCE/connection.json $OUTPUT/connection.json
+
+exit 0
+
+```
+
+### /bin/release
+
+The `bin/release script` is responsible for providing metadata chaincode to the peer. For chaincode as an external service, the `bin/release` script is responsible for providing the connection.json to the peer by placing it in the `RELEASE_OUTPUT_DIR`. The peer invokes the release script with two arguments:
+
+```
+bin/release BUILD_OUTPUT_DIR RELEASE_OUTPUT_DIR
+```
+
+A typical `release` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+set -x
+
+if [ "$#" -ne 2 ]; then
+    >&2 echo "Expected 2 directories got $#"
+    exit 2
+fi
+
+BLD="$1"
+RELEASE="$2"
+
+#external chaincodes expect artifacts to be placed under "$RELEASE"/chaincode/server
+if [ -f $BLD/connection.json ]; then
+   mkdir -p "$RELEASE"/chaincode/server
+       cp $BLD/connection.json "$RELEASE"/chaincode/server
```

Comment:
Maybe a tab instead of spacing? But misallignment.

Reply:
Done